apps/emqx_auth_jwt/src/emqx_auth_jwt.erl

```diff
@@ -109,11 +109,17 @@ string_to_number(_) ->
 %% Verify Claims
 %%--------------------------------------------------------------------
 
-verify_acl(ClientInfo, #{<<"sub">> := SubTopics}, subscribe, Topic) when is_list(SubTopics) ->
-    verify_acl(ClientInfo, SubTopics, Topic);
-verify_acl(ClientInfo, #{<<"pub">> := PubTopics}, publish, Topic) when is_list(PubTopics) ->
-    verify_acl(ClientInfo, PubTopics, Topic);
-verify_acl(_ClientInfo, _Acl, _PubSub, _Topic) -> {stop, deny}.
+verify_acl(ClientInfo, Acl, PubSub, Topic) ->
+    Key = case PubSub of
+              subscribe -> <<"sub">>;
+              publish -> <<"pub">>
+          end,
+    case {maps:get(<<"all">>, Acl, []), maps:get(Key, Acl, [])} of
+        {Rules1, Rules2} when is_list(Rules1), is_list(Rules2) ->
+            verify_acl(ClientInfo, Rules1 ++ Rules2, Topic);
+        {_, _} ->
```

Comment:
when does it end up here?

Reply:
When the format of one of the `all` or `pub`, `sub` rules is incorrect? i.e:
```
{ "sub": ["t/a", "t/b"],
  "all": "bad_format_here"
}
```